### PR TITLE
Do not nuke existing folders in git checkout.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

I did not enjoy accidentally using a path with files I cared about ;(
  
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
I think the class description needs to be updated as well. https://github.com/opensearch-project/opensearch-build/pull/248/files#diff-667ead02eb7cd0e63a1df908d4cc8599c54def808ce002a4fc73b1f5f06ccab2R11

File: bundle-workflow/python/git/git_repository.py [hunk 1]
```diff
@@ -19,9 +19,7 @@ def __init__(self, url, ref, directory = None):
             self.dir = self.temp_dir.name
         else:
             self.dir = directory
-            print(f'Deleting {self.dir}')
```
Comment (peternied) [inline bundle-workflow/python/git/git_repository.py:22]: I think there was utility in being sure the folder was cleaned up? Maybe we add a default that causes this to barf unless you've set `allow-nuke=true`

**Review by peternied — APPROVED** [review]

File: bundle-workflow/python/git/git_repository.py (hunk 1 above)
Comment (dblock) [reply bundle-workflow/python/git/git_repository.py:22]: We checkout into a temporary folder, so these actually do get cleaned up. Making GitRepository act like a temporary folder itself seems unnecessary, but I would be fine if we needed that. The problem here is that this class is cleaning up something it didn't create!

**Comment by dblock** [comment]
> I think the class description needs to be updated as well

Done.

**Review by mch2 — APPROVED** [review]

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**